### DISCUSSION TITLE
How mydumper/myloader handles native mysqldump/mysql error codes

### DISCUSSION THREAD

**Discussion by saifalicw** [mydumper/mydumper#1443] [Q&A] [answered]
We are working on a feature that uses **mydumper/myloader** but we are stuck at a use case where we were previously handling some mysql error codes, but we are not sure how **mydumper/myloader** outputs errors to stdout, for instance let's talk about.

```
MySQLResult="$(myloader --database $$Dest --directory $DumpPath/$Dumpfile --threads ${threads})"

if [[ $MySQLResult =~ .*ERROR.*1049.* ]]; then
    # Apply Patch
fi

if [[ ${MySQLResult^^} =~ .*ERROR.*1118.*(BLOB.*TEXT|TEXT.*BLOB).* ]]; then
    # Apply Patch
fi

if [[ $MySqlDump =~ .*mysqldump.*Got.*error.*144:.* ]]; then
    # Apply patch
fi
```

Please share some insights, if myloader/mydumper will return the exit code or it will through some output.

**Comment by davidducos** [comment, marked as answer]
Hi @saifalicw, 
If there is any error during import, then myloader will return a non zero value. 
In the log (that you can get setting -L), you will find the entry with the error code.

**Reply by saifalicw** [reply]
Hello @davidducos, Thanks for the response.
So that means our above conditions will pick the error code already or we need to do some parsing separately for **myloader**?
